tasks/main.yml

```diff
@@ -29,7 +29,10 @@
     - '{{ etcd_cluster_pki_dir }}'
 
 - include_tasks: pki.yml
-  when: etcd_secure
+  when: etcd_secure and not etcd_gen_certs
+
+- include_tasks: gen-pki.yml
+  when: etcd_secure and etcd_gen_certs
```

Comment:
If I read `pki.yml` right, it looks like the keys would be expected to exist on the controller at a certain location.  So feels like we could stat those local files to see if they exists, and use that to determine whether to generate the certs, rather than introducing a new var here.